### ISSUE TITLE
CI failure labeling

### ISSUE THREAD

**Issue by fjetter** [coiled/coiled-runtime#224]
I believe it would be very valuable to label the failed CI tickets to see how we are doing over time. I think the most interesting bit would be to distinguish whether this has been an internal failure (e.g. CI pipeline broken, test logic false) an upstream dask issue (e.g. API breakage) or a platform issue (i.e. Coiled specific).

- `cause: upstream`
- `cause: platform`
- `cause: internal`

Our process should be adjusted to add a label to the ticket before closing.

Thoughts? Is there anything else worth labeling?

cc @ncclementi @jrbourbeau

**Comment by ncclementi** [comment]
This sounds good to me!

**Comment by jrbourbeau** [comment]
Thanks for the idea @fjetter -- I agree with you and @ncclementi that this would be nice to do. I've gone ahead and added labels for this. The only change I made was I added two upstream labels, `cause: upstream (dask)` and `cause: upstream (other)`, for Dask and non-Dask related failures. My thinking is it'll be useful to disentangle, for example, a breaking Dask API change from a change in `fsspec` or some transient conda-forge HTTP error. Certainly open to thoughts from others and happy to iterate
